backfilling-data.md

```diff
@@ -0,0 +1,91 @@
+```
+
+You should now see a set of files in `/data/versionista-backfill` named like `pages-versionista1-0.json` with increasing numbers on the end. Make sure there were no errors (output in files named `errors-versionista1.log`).
+
+
+## Scrape matching versions from candidate pages (repeat for each chunk)
+
+Check the list of files now in `/data/versionista-backfill` to see how many chunks you have to go through. Then repeat the following steps once for each chunk. Note they all use the `$CHUNK` variable created in the first line, so the only line you need to change each time is the first (set it to the number of the chunk you are running).
+
+This will pull down the actual raw page content, diffs, and version metadata for all the pages in a chunk.
+
+**Make sure to set the `--after` and `--before` options the same as you did for `get-versionista-metadata` above.**
+
+```sh
+# Increment this variable each time until you've done all the chunks
+CHUNK=0
+# Actually scrape version metadata, raw content, and diffs
+./bin/get-versionista-page-chunk --after '2017-10-09T00:00:00Z' --before '2017-10-18T00:00:00Z' --relative-paths /data/versionista-backfill/ --save-content --save-diffs --parallel 3 --pause-time 10000 --format json-stream --output /data/versionista-backfill/$VERSIONISTA_NAME/metadata-chunk-$CHUNK.json --errors /data/versionista-backfill/$VERSIONISTA_NAME/errors-page-chunk-$CHUNK.log --candidate-pages /data/versionista-backfill/pages-$VERSIONISTA_NAME-$CHUNK.json
+# Add a newline to the end of the file (should fix our JSON serialization to do this)
+echo '' >> /data/versionista-backfill/$VERSIONISTA_NAME/metadata-chunk-$CHUNK.json
+```
+
+You should now have a file and directory structure in `/data/versionista-backfill/versionista1` that resembles what you’d get when running the normal `scrape-versionista` script:
+
+```
+/data/versionista-backfill
+└─┬ /versionista1
+  ├─┬ /96855-7670814
+  │ ├── diff-13191115.html
+  │ ├── diff-13191115-text.html
+  │ ├── version-13191115.html
+  │ └── # etc
+  ├─┬ /96855-7670957
+  │ ├── diff-13191115.html
+  │ ├── diff-13191115-text.html
+  │ ├── version-13191115.html
+  │ └── # etc
+  ├── /{site_id}-{page_id}  # etc etc etc
+  ├── metadata-chunk-0.json
+  ├── metadata-chunk-1.json
+  └── metadata-chunk-{chunk_number}.json  # etc etc etc
+```
+
+Follow all this up by combining the above metadata files into a single one for importing to web-monitoring-db:
+
+```sh
+cat /data/versionista-backfill/$VERSIONISTA_NAME/metadata-chunk-* > /data/versionista-backfill/$VERSIONISTA_NAME/metadata.json
+```
+
+The rest of this process is a manual version of what the `scrape-versionista-and-upload` script does.
+
+
+## Upload to cloud storage
```

Comment:
I didn't do any of the uploading because I don't have any servers setup. But looks like I shouldn't *\*fingers crossed\** need to if this is just manually doing what's always been done and nothing has changed. If you want me to try these steps as well, I'll have to do it sometime this weekend to get everything setup properly.